superset-frontend/src/views/CRUD/utils.tsx

```diff
@@ -167,8 +167,8 @@ export const getUserOwnedObjects = (
   const filters = {
     created: [
       {
-        col: 'created_by',
-        opr: 'rel_o_m',
+        col: resource === 'saved_query' ? 'created_by' : 'owners',
```

Comment:
this kind of defeats the purpose to me, and makes this function harder to reuse.
I would add a `col` and `opr` as third and fourth param, assigning the 'owners' and 'rel_m_m' as defaults.
Then, when calling for `saved_query`, we can override those, and reuse it for other calls if needed.